src/vm/methodtable.cpp

```diff
@@ -10244,7 +10244,14 @@ static BOOL ComputeIsLayoutInCurrentVersionBubble(MethodTable* pMT)
     ApproxFieldDescIterator fieldIterator(pMT, ApproxFieldDescIterator::INSTANCE_FIELDS);
     for (FieldDesc *pFD = fieldIterator.Next(); pFD != NULL; pFD = fieldIterator.Next())
     {
-        MethodTable * pFieldMT = pFD->GetApproxFieldTypeHandleThrowing().AsMethodTable();
+        TypeHandle thField = pFD->GetApproxFieldTypeHandleThrowing();
+
+        MethodTable * pFieldMT = NULL;
+        if (thField.IsTypeDesc())
```

Comment:
Just call `thField.GetMethodTable()` directly. You do not need `thField.IsTypeDesc()` condition here, etc.

Reply:
Thanks for poining this API out. That's exactly what I needed :) I had no idea we already had it. I'll make that change